webstepper/src/App.tsx

```diff
@@ -20,6 +20,23 @@ export default function App() {
         setStep(Math.min(Math.max(newStep, 0), limit - 1));
     };
 
+    useEffect(() => {
+        const handleKeyDown = (event) => {
+            if (event.key == "ArrowLeft") {
+                handleStep(step - 1);
+            }
+            if (event.key == "ArrowRight") {
+                handleStep(step + 1);
+            }
+        };
+
+        document.addEventListener("keydown", handleKeyDown);
+
+        return () => {
+            document.removeEventListener("keydown", handleKeyDown);
+        };
+    }, [step]);
```

Comment:
I think it's actually somewhat inefficient here to use `[step]` as a dependency just to pass the right value to `handleStep`. A better approach is to:

1. Modify the `handleStep` function above to take an *offset* rather than the new step value (note that inside `setStep`, you can pass an [*updater function*](https://react.dev/reference/react/useState#updating-state-based-on-the-previous-state) that takes the current state value and returns the new state value.
2. Change this dependency list to just `[]`, so that this is only run after the initial render.